app/src/main/java/appricottsoftware/clarity/HomeActivity.java

```diff
@@ -45,11 +64,20 @@
     private static SettingFragment settingFragment;
     private static PlayerFragment playerFragment;
 
+    private Intent playerServiceIntent;
+    private MediaBrowserCompat mediaBrowser;
+    private MediaBrowserCompat.ConnectionCallback connectionCallback;
+    private MediaControllerCompat.Callback controllerCallback;
+    private MediaControllerCompat mediaController;
+
+    private Context context;
```

Comment:
Why get context when you can you 'this'

Reply:
Runnable is an inner class that can't access 'this'.